### PR TITLE
State serialization to 0.5.0

### PR DESCRIPTION
**Pull request by carlos-romano**

### CODE REVIEW

File: internal/service/service.go [hunk 1]
```diff
@@ -82,6 +83,18 @@ type PreImageMetaKey struct {
 }
 type PreimageHistoricalTimeslots []jamtime.Timeslot // Metadata for preimages (l) - TODO: There is a MaxHistoricalTimeslotsForPreimageMeta.
 
+func (p PreimageHistoricalTimeslots) ToByteSlice() []byte {
+	// Allocate a byte slice large enough to hold all Timeslot values
+	output := make([]byte, len(p)*4)
+
+	// Convert each Timeslot to bytes and copy into the output slice
+	for i, ts := range p {
```
Comment (pantrif) [inline internal/service/service.go:91]: I think you should use jam codec for encoding timeslot

File: internal/state/merkle/serialization_utils_test.go [hunk 2]
```diff
@@ -10,25 +12,63 @@ import (
...
-	binary.BigEndian.PutUint32(expectedEncodedServiceId, uint32(serviceId))
-	assert.Equal(t, expectedEncodedServiceId, stateKey[1:5])
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Generate the state key
+			stateKey := generateStateKeyInterleavedBasic(tt.i, tt.serviceId)
+
+			// Convert serviceId to bytes for verification
+			serviceIdBytes := make([]byte, 4)
+			binary.BigEndian.PutUint32(serviceIdBytes, uint32(tt.serviceId))
```
Comment (pantrif) [inline internal/state/merkle/serialization_utils_test.go:46]: Is there any reason why you use big endian encoding? I think you should use jam integer encoding here (little endian)

Comment (carlos-romano) [reply internal/state/merkle/serialization_utils_test.go:46]: If you see the implementation of generateStateKeyInterleavedBasic, we need that ordering. With Little Endian won't pass

**Review by danielvladco — DISMISSED** [review]

**Review by greywolve — DISMISSED** [review]

File: internal/state/merkle/serialization_utils_test.go (hunk 2 above)
Comment (carlos-romano) [reply internal/state/merkle/serialization_utils_test.go]: Agreed that the implmentation should be different. Updated

**Review by pantrif — APPROVED** [review]

**Review by pantrif — APPROVED** [review]

**Pull request merged**